backend/kafka_backend/kafka_backend.go

```diff
@@ -0,0 +1,131 @@
+package kafkabackend
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/skroutz/downloader/job"
+)
+
+// FlushTimeout is the timeout we give to our kafka producer
+// to flush pending messages.
+const FlushTimeout = 5000
+
+// Backend notifies about a job completion by producing to a Kafka topic.
+type Backend struct {
+	producer *kafka.Producer
+	reports  chan job.Callback
+	eventsWg *sync.WaitGroup
+}
+
+// ID returns "kafka".
+func (b *Backend) ID() string {
+	return "kafka"
+}
+
+// Start starts the backend by creating a producer,
+// given a set of options provided by the configuration.
+func (b *Backend) Start(ctx context.Context, cfg map[string]interface{}) error {
+	var err error
+
+	kafkaCfg := make(kafka.ConfigMap)
+	for k, v := range cfg {
+		err := kafkaCfg.SetKey(k, v)
+		if err != nil {
+			return err
+		}
+	}
+
+	b.producer, err = kafka.NewProducer(&kafkaCfg)
+	if err != nil {
+		return err
+	}
+
+	b.reports = make(chan job.Callback)
+	b.eventsWg = new(sync.WaitGroup)
+
+	// start a go routine to monitor Kafka's Events channel
+	b.eventsWg.Add(1)
+	go func() {
+		b.transformStream(ctx)
+		b.eventsWg.Done()
+	}()
+
+	return nil
+}
+
+// Notify produces (asynchronously) a payload message into a kafka topic using
+// the producer. Kafka will send the delivery report to the default Events delivery channel.
+func (b *Backend) Notify(destination string, cbInfo job.Callback) error {
```

Comment:
Consider renaming `destination` to something more informative and shorter: `topic`.